src/main/java/spacesheepgames/gocd/gcs/artifact/plugin/executors/PublishArtifactExecutor.java

```diff
@@ -60,6 +60,7 @@ public GoPluginApiResponse execute() {
         ArtifactPlan artifactPlan = publishArtifactRequest.getArtifactPlan();
         final ArtifactStoreConfig artifactStoreConfig = publishArtifactRequest.getArtifactStore().getArtifactStoreConfig();
         try {
+            final int bufferSize = 1024 * 1024 * 128; // 128Mi
```

Comment:
Um ponto que está me preocupando é usar esse buffer grande quando tem muitos arquivos. Não sei dizer como a biblioteca funciona por trás dos panos, mas imagino que a cada arquivo ele vai alocar o tamanho do buffer. 
Como ele roda em um `for` de forma síncrona acho que não vai ter erro de memória.

As alternativas que eu vejo seria:
1. Ajustar o tamanho do buffer para atender arquivos grandes e vários pequenos
2. Ter o tamanho de buffer diferente para um vs muitos arquivos